backend/src/apiserver/resource/resource_manager.go

```diff
@@ -493,58 +493,72 @@ func (r *ResourceManager) DeleteJob(jobID string) error {
-	}
-	runDetail := &model.RunDetail{
-		Run: model.Run{
-			UUID:             runId,
-			DisplayName:      workflow.Name,
-			Name:             workflow.Name,
-			StorageState:     api.Run_STORAGESTATE_AVAILABLE.String(),
-			Namespace:        workflow.Namespace,
-			CreatedAtInSec:   workflow.CreationTimestamp.Unix(),
-			ScheduledAtInSec: workflow.ScheduledAtInSecOr0(),
-			FinishedAtInSec:  workflow.FinishedAt(),
-			Conditions:       workflow.Condition(),
-			PipelineSpec: model.PipelineSpec{
-				WorkflowSpecManifest: workflow.GetWorkflowSpec().ToStringForStore(),
-			},
-			ResourceReferences: []*model.ResourceReference{
-				{
-					ResourceUUID:  runId,
-					ResourceType:  common.Run,
-					ReferenceUUID: jobId,
-					ReferenceType: common.Job,
-					Relationship:  common.Creator,
+		err := r.runStore.UpdateRun(runId, workflow.Condition(), workflow.FinishedAt(), workflow.ToStringForStore())
+		if err != nil {
+			return util.Wrap(err, "Failed to update the run.")
+		}
+	} else {
+		// Get the experiment resource reference for job.
+		experimentRef, err := r.resourceReferenceStore.GetResourceReference(jobId, common.Job, common.Experiment)
+		if err != nil {
+			return util.Wrap(err, "Failed to retrieve the experiment ID for the job that created the run.")
+		}
+		runDetail := &model.RunDetail{
+			Run: model.Run{
+				UUID:             runId,
+				DisplayName:      workflow.Name,
+				Name:             workflow.Name,
+				StorageState:     api.Run_STORAGESTATE_AVAILABLE.String(),
+				Namespace:        workflow.Namespace,
+				CreatedAtInSec:   workflow.CreationTimestamp.Unix(),
+				ScheduledAtInSec: workflow.ScheduledAtInSecOr0(),
+				FinishedAtInSec:  workflow.FinishedAt(),
+				Conditions:       workflow.Condition(),
+				PipelineSpec: model.PipelineSpec{
+					WorkflowSpecManifest: workflow.GetWorkflowSpec().ToStringForStore(),
 				},
-				{
-					ResourceUUID:  runId,
-					ResourceType:  common.Run,
-					ReferenceUUID: experimentRef.ReferenceUUID,
-					ReferenceType: common.Experiment,
-					Relationship:  common.Owner,
+				ResourceReferences: []*model.ResourceReference{
+					{
+						ResourceUUID:  runId,
+						ResourceType:  common.Run,
+						ReferenceUUID: jobId,
+						ReferenceType: common.Job,
+						Relationship:  common.Creator,
+					},
+					{
+						ResourceUUID:  runId,
+						ResourceType:  common.Run,
+						ReferenceUUID: experimentRef.ReferenceUUID,
+						ReferenceType: common.Experiment,
+						Relationship:  common.Owner,
+					},
 				},
 			},
-		},
-		PipelineRuntime: model.PipelineRuntime{
-			WorkflowRuntimeManifest: workflow.ToStringForStore(),
-		},
+			PipelineRuntime: model.PipelineRuntime{
+				WorkflowRuntimeManifest: workflow.ToStringForStore(),
+			},
+		}
+		err = r.runStore.CreateOrUpdateRun(runDetail)
+		if err != nil {
+			return util.Wrap(err, "Failed to create or update the run.")
+		}
+	}
+
+	if workflow.IsInFinalState() {
```

Comment:
Does it mean it will delete all finished workflow immediately? It's better to make it to be configurable so user can debug argo workflow after it's finished.